scripts/detect_artefacts.py

```diff
@@ -0,0 +1,311 @@
+                "rpn_nms_thresh": 0.2,
+                "rpn_batch_size_per_image": 250
+                }
+    kwargs = {**defaults}
+
+    model = torchvision.models.detection.__dict__[args.arch](pretrained=False, num_classes=5, **kwargs)
+    model.load_state_dict(state_dict, strict=False)
+    model.roi_heads.box_roi_pool = MultiScaleRoIAlign(featmap_names=['0', '1', '2', '3'], output_size=(7, 7),
+                                                      sampling_ratio=2)
+    anchor_sizes = ((16), (64), (128), (264))
+    aspect_ratios = ((0.5, 1.0, 2.0, 3.0,)) * len(anchor_sizes)
+    model.rpn.anchor_generator.sizes = anchor_sizes
+    model.rpn.anchor_generator.aspect_ratios = aspect_ratios
+    # GPU
+    if isinstance(args.device, int):
+        if not torch.cuda.is_available():
+            raise AssertionError("PyTorch cannot access your GPU. Please investigate!")
+        if args.device >= torch.cuda.device_count():
+            raise ValueError("Invalid device index")
+    # Silent default switch to GPU if available
+    elif torch.cuda.is_available():
+        args.device = 0
+    else:
+        logging.warning("No accessible GPU, target device set to CPU.")
+    if torch.cuda.is_available():
+        torch.cuda.set_device(args.device)
+        model = model.cuda()
+    root_dir = args.root_dir
+    save_to_folder = args.save_to
+    checkpoint_path = args.checkpoint_path
+    save_image = args.plot
+    code_KIE = args.KIE
+    save_j = args.save_json
+    if_save = args.json
+    model.load_state_dict(torch.load(checkpoint_path))
+    model.eval()
+    test_list = os.listdir(args.root_dir)
+    inf_dic = inference_script(model, root_dir, test_list, code_KIE,
+                               save_image, save_to_folder)
+    if if_save:
+        with open(save_j, 'w', encoding='utf-8') as f:
+            json.dump(inf_dic, f)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Artefact Detection Training",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('root_dir', type=str, help='path to image folder')
+    parser.add_argument('checkpoint_path', type=str, help='path to torch checkpoint')
+    parser.add_argument('-save_to', type=str, required=False,
+                        help="Use this along with --plot True to save images to the desired location")
+    parser.add_argument('-save_json', type=str, required=False,
+                        help="Use this along with --json True")
+    parser.add_argument('--json', nargs='?', const=False, type=bool,
+                        help="Optional: If user wants to save json output")
+    parser.add_argument('--KIE', nargs='?', const=False, type=bool,
+                        help="Optional:If user wants to extract information from codes")
+    parser.add_argument('--plot', nargs="?", const=False, type=bool,
+                        help="Optional:If the user wants to plot and save the predictions")
```

Comment:
what are nargs?

Reply:
I used them initially as the use of nargs helps us to input a variety of default values in different types. But as we dont need that right now, I have changed it :)